public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -438,8 +438,13 @@ export class LokiDatasource
   }
 
   async metadataRequest(url: string, params?: Record<string, string | number>) {
-    const res = await lastValueFrom(this._request(url, params, { hideFromInspector: true }));
-    return res.data.data || res.data.values || [];
+    if (config.featureToggles.lokiBackendMode) {
+      const res = await this.getResource(url, params);
+      return res.data || res.values || [];
```

Comment:
As we talked about, `res.values` can be probably removed as this was used only before v1 of loki.

Reply:
i did some digging and things are somewhat more complicated, i recommend we do this change in a separate PR.